### PR TITLE
AP_HAL: Remove memory leaks

### PR DESCRIPTION
**Pull request by KimHyungSub**
Issues:#8642, #8644 
I correct these issues and then open a pull request.

### CODE REVIEW

File: libraries/AP_HAL_Linux/benchmarks/benchmark_videoin.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ static void BM_Crop8bpp(benchmark::State& state)
     new_buffer = (uint8_t *)malloc(state.range_x() * state.range_y());
     if (!new_buffer) {
         fprintf(stderr, "error: couldn't malloc new_buffer\n");
+		free(buffer);
```
Comment (rmackay9) [inline libraries/AP_HAL_Linux/benchmarks/benchmark_videoin.cpp]: tiny thing but there's a missing space .. might as well make it lineup.

**Comment by rmackay9** [comment]
I'm not really qualified to review this but as a style thing, the commit should start with "AP_HAL:".  It's possible it should be split into two commits but I'll leave that for others to decide.

**Comment by KimHyungSub** [comment]
@rmackay9  If you want to split this commit into two commits, I will do that.
Please close this pull request.

**Comment by tridge** [comment]
I have split up the commits and fixed the whitespace 
thanks for the fixes!

**Comment by KimHyungSub** [comment]
@tridge  Oh, Thank you for the commits!

**Comment by OXINARF** [comment]
Slightly changed commit messages (so that each references the appropriate issue) and merged, thanks for your first contribution @KimHyungSub!

By the way, if you haven't read our contribution rules (http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html), it would be nice if you did.

**Pull request closed without merging**